server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -696,6 +696,32 @@ static String getCommitCheckpointFileName(long generation) {
         return TRANSLOG_FILE_PREFIX + generation + CHECKPOINT_SUFFIX;
     }
 
+    /**
+     * Trims translog for terms of files below <code>belowTerm</code> and seq# above <code>aboveSeqNo</code>.
+     * Effectively it moves max visible seq# {@link Checkpoint#trimmedAboveSeqNo} therefore {@link TranslogSnapshot} skips those operations.
+     */
+    public void trim(long belowTerm, long aboveSeqNo) throws IOException {
+        assert aboveSeqNo > 0 : "aboveSeqNo has to be positive number";
+        try (ReleasableLock lock = writeLock.acquire()) {
+
```

Comment:
can you check the term of the current file is >= belowTerm and thus doesn't need to be trimmed? if throw an IlegalArgumentException.